### ISSUE TITLE
Proposal: Replace CheckBox "Tick" glyph to look better

### ISSUE THREAD

**Issue by mdtauk** [microsoft/microsoft-ui-xaml#989]
# Proposal: Replace CheckBox "Tick" glyph to look better

## Summary
The Tick glyph used for the CheckBox control can seem a little thin and too close to the edges of the box.  FastDNA has the nicest looking version to my eye, compared to the other Microsoft UI frameworks - and I would like to see the others changed to match.  The Indeterminate mark and the Checked mark, feel more in balance

----

![image](https://user-images.githubusercontent.com/7389110/60627291-7eaf4c80-9de5-11e9-8aec-0fe3889da40d.png)
_FastDNA CheckBox_

----

![image](https://user-images.githubusercontent.com/7389110/60627029-933f1500-9de4-11e9-87ee-6bddf10c6e9d.png)
_Fabric Web_

----

![image](https://user-images.githubusercontent.com/7389110/60627364-bddd9d80-9de5-11e9-9277-c5cd107015e5.png)
_XAML UWP_

----

![image](https://user-images.githubusercontent.com/7389110/60627423-f2e9f000-9de5-11e9-9aa0-010ac6b29336.png)
_Office UWP_

----

![image](https://user-images.githubusercontent.com/7389110/60627457-0ac17400-9de6-11e9-91c9-8be152286166.png)
_Win32_

----

![image](https://user-images.githubusercontent.com/7389110/60627579-61c74900-9de6-11e9-98f6-ef46c0670cd2.png)
_Material Design - Android_

----

![image](https://user-images.githubusercontent.com/7389110/60627628-84f1f880-9de6-11e9-9c81-66cabbe904b1.png)
_macOS_

## Important Notes
FastDNA uses an SVG shape, but either the current glyph in the MDL2 font, or a new altered glyph should be added to match the look.

![image](https://user-images.githubusercontent.com/7389110/60627660-a652e480-9de6-11e9-9498-5c1e3aff5dc7.png)

**Comment by jevansaks** [comment]
Thanks @mdtauk! There is a proposal #839 which will include an update to the checkbox visual. Looks like you've weighed in on that proposal, is there more in this suggestion that isn't captured in the proposed updates?

**Comment by mdtauk** [comment]
@jevansaks  I understood that proposal to be focused on ensuring the CheckBox used in various WinUI controls, are consistent amongst themselves.  Also that the proposed visual comps, continues to use the current MDL2 glyph, with only a change of fontsize where needed.

I am happy for this proposal to be rolled into that one, if there is scope to change the checkmark itself.

**Comment by jevansaks** [comment]
I see, yes then I think these should be separate proposals. @chigy do you know if a checkmark glyph update is part of this design update?

**Comment by shaheedmalik** [comment]
The filled in ones look the best.

Maybe like this instead?

![image](https://user-images.githubusercontent.com/51347950/60690408-45ceb080-9e95-11e9-99c6-e64581eff858.png)

**Comment by Poopooracoocoo** [comment]
@mdtauk The win32 screenshot is from Windows 7. Mind updating it to one from W10?

I really dislike the new checkboxes in #839 even though I want checkbox consistency. It's the only reason I reacted with a thumbs down

**Comment by mdtauk** [comment]
@Poopooracoocoo  Is this image better?

![image](https://user-images.githubusercontent.com/7389110/60708340-c200dc80-9f05-11e9-855d-7767f4c561c4.png)

----

@shaheedmalik  The proposal is only about the design of the checkmark, not the full control - but filled on Checked and Indeterminate states makes sense to me, and fits in with most modern UI approaches from other OSs.

Here is the same checkmark applied to the UWP style checkbox states

![image](https://user-images.githubusercontent.com/7389110/60711745-718d7d00-9f0d-11e9-968d-eaf80ef0187c.png)

**Comment by mdtauk** [comment]
I know within the MDL 2 Assets font, there are checkbox glyphs.  I see both 16 px and 14 px scaled glyphs, and so I made a quick font with 20px, 16px, and 14px scaled versions of the new checkmark, but rounded to match the upcoming control designs.

[WinUI.zip](https://github.com/microsoft/microsoft-ui-xaml/files/3362303/WinUI.zip)
_TTF file and FontLab editable file_

**Comment by chigy** [comment]
@mdtauk , I don't have time to look deeply into this request yet but by just scanning, it looks like a duplicate request to issue #839?

**Comment by mdtauk** [comment]
> @mdtauk , I don't have time to look deeply into this request yet but by just scanning, it looks like a duplicate request to issue #839?

@chigy  It is somewhat related, but it is specifically about replacing the current "tick" glyph, with the one used by FastDNA's CheckBoxes

**Comment by bheston** [comment]
Thanks for the comments on the FAST check icon. We're actually working right now on switching the checked state to a filled style. That icon will remain the same shape, but will be white over a neutral fill.

The indeterminate mode is being updated to a dash more than a square, to further differentiate it from a radio button. It will be unfilled.

This coordinates with the platform / user agent changes we've worked on with Chromium, which will be rolling out shortly.

**Comment by mdtauk** [comment]
@bheston  could you share the check mark SVG with the Windows team, so the thicker shape can be used in WinUI and Xbox

**Comment by bheston** [comment]
Here it is.
[fast-check.svg.txt](https://github.com/microsoft/microsoft-ui-xaml/files/3623648/fast-check.svg.txt)

And the indeterminate one we are updating to.
[fast-indeterminate.svg.txt](https://github.com/microsoft/microsoft-ui-xaml/files/3623659/fast-indeterminate.svg.txt)

Note that indeterminate will be unfilled, since clicking it in this state toggles it to checked.
![image](https://user-images.githubusercontent.com/47367562/65084224-76d06480-d95f-11e9-8da3-104edd07c308.png)

**Comment by chigy** [comment]
Folks often ask us about the alignment between WinUI and Fabric. This one is actually a case where we are in alignment. FastDNA cannot use the MDL2 font.

![image](https://user-images.githubusercontent.com/22958680/66614339-8a957200-eb7d-11e9-89f9-a8b4e505e20f.png)

**Comment by mdtauk** [comment]
It is true that both Fabric and WinUI are using the same glyph.  My point was that I felt the Checkmark used by FastDNA was a better glyph to use.  It has a thicker stroke, which fits better with the Indeterminate/ThreeState glyph, as well as the Radio Button glyph.

So if adding a new thicker tick mark to the MDL2 font is how this "problem" is solved, then that could be done.

**Comment by chigy** [comment]
> It is true that both Fabric and WinUI are using the same glyph. My point was that I felt the Checkmark used by FastDNA was a better glyph to use. It has a thicker stroke, which fits better with the Indeterminate/ThreeState glyph, as well as the Radio Button glyph.
> 
> So if adding a new thicker tick mark to the MDL2 font is how this "problem" is solved, then that could be done.

@mdtauk , yes I understood where you were coming from but I was also commenting on some other feedback about alignment issues. This has to go to both Fabric and Windows design team to see if they want to thicken the tick glyph.

Can you please update your proposal to be very clear "Make Tick glyph thicker" or something like that? I can do that, too, if it is agreeable.

**Comment by mdtauk** [comment]
> @mdtauk , yes I understood where you were coming from but I was also commenting on some other feedback about alignment issues. This has to go to both Fabric and Windows design team to see if they want to thicken the tick glyph.
> 
> Can you please update your proposal to be very clear "Make Tick glyph thicker" or something like that? I can do that, too, if it is agreeable.

I updated the title - I originally mentioned FastDNA as it, like the Office XAML tick mark, uses a different design, that I assume was approved before appearing, so just making an arbitrary change, might not be deemed appropriate.

**Comment by niels9001** [comment]
@mdtauk feels like this issue is pretty stale and can be closed, right?

**Comment by mdtauk** [comment]
> [@mdtauk](https://github.com/mdtauk) feels like this issue is pretty stale and can be closed, right?

I suppose so if there is no intention to take a look at the design again, but I believe the thinness of the checkmark and indeterminate mark, is still hampering the visibility, and the radio button is better as the mark is more visible

<img width="111" height="115" alt="Image" src="https://github.com/user-attachments/assets/6e3ad0d1-177a-4d3c-abe1-292ab55453a6" />

<img width="266" height="99" alt="Image" src="https://github.com/user-attachments/assets/b8062ea8-2044-4a3a-a3a5-f4d25f65158a" />

There is still a difference compared to Fluent 2 also

<img width="291" height="276" alt="Image" src="https://github.com/user-attachments/assets/85f217df-2b51-4014-bb2c-423d5d4f75bd" />
